api/python/provisioner/commands/validator/validator.py

```diff
@@ -0,0 +1,252 @@
+
+    Attributes
+    ----------
+    _required: bool
+        if True validation raises an Exception if the file doesn't exist
+    _content_validator: Callable[[Path], None]
+        callable object for file content validation.
+
+        Should raise the `ValidationError` exception if file content validation
+        is failed
+
+    Notes
+    -----
+    TBD: other possible parameters
+        permissions: int/str
+            requested file permissions
+        owner: str
+            requested file owner
+        group: str
+            requested file group
+        is_symlink: bool
+            defines if file should be a symlink
+
+    At initialization time `attr.s` is responsible for `_required`
+    and `_content_validator` attributes validation.
+    """
+
+    _required: bool = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(bool)),
+        default=False
+    )
+    _content_validator: Callable[[Path], None] = attr.ib(
+        validator=attr.validators.optional(attr.validators.is_callable()),
+        default=None,
+    )
+
+    def validate(self, path: Path):
+        """
+        Validate the file by a given path.
+
+        Parameters
+        ----------
+        path: Path
+            path for file validation
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValidationError
+            If validation is failed.
+
+        """
+        logger.debug(f"Start '{path}' file validation")
+
+        if not path.exists():
+            if self._required:
+                raise ValidationError(reason=f"File '{path}' should exist.")
+            return
+
+        logger.debug(f"File '{path}' exists.")
+
+        if not path.is_file():
+            raise ValidationError(reason=f"'{path}' is not a regular file.")
+
+        if self._content_validator and callable(self._content_validator):
+            # Should raise an Exception if validation fails
+            self._content_validator(path)
+
+
+@attr.s(auto_attribs=True)
+class DirValidator(Validator):
+
+    """
+    Class for catalog validation.
+
+    Attributes
+    ----------
+    _files_scheme: Dict
+        Nested catalog structure for the validation path
+
+    _required: bool
+        if True validation raises an Exception if the directory
+        doesn't exist
+
+    Notes
+    -----
+    TBD: other possible parameters
+    permissions: int/str
+        requested directory permissions
+    owner: str
+        requested directory owner
+    group: str
+        requested directory group
+
+    At initialization time `attr.s` is responsible for `_file_scheme`
+    validation and converting keys of file scheme from string to `Path`
+    instance.
+    """
+
+    _files_scheme: Dict = attr.ib(
+        validator=attr.validators.optional(
+            attr.validators.deep_mapping(
+                key_validator=attr.validators.instance_of((str, Path)),
+                # we can't refer to DirValidator here.
```

Comment:
Originally, it was supposed to be like
```python
value_validator=attr.validators.instance_of((FileValidator, DirValidator)),
```
Since I started developing the new Validators for validation, it is better to leave only parent class here.